src/backend/libc/fs/syscalls.rs

```diff
@@ -1734,10 +1733,13 @@ pub(crate) fn fallocate(
     offset: u64,
     len: u64,
 ) -> io::Result<()> {
-    // Silently cast; we'll get `EINVAL` if the value is negative.
+    // Silently cast to `i64`; we'll get `EINVAL` if the value is negative.
     let offset = offset as i64;
     let len = len as i64;
 
+    // ESP-IDF and Vita don't support 64-bit offsets, for example.
+    let offset = offset.try_into().map_err(|_| io::Errno::OVERFLOW)?;
+
```

Comment:
A cast is needed for `len` here too.

Reply:
(I failed to point this out in #1432; it is the only non-`offset` case that I personally have run into, but there may be others I missed.)